core/src/main/java/jenkins/model/ParameterizedJobMixIn.java

```diff
@@ -215,6 +215,9 @@ public final void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp,
         hudson.model.BuildAuthorizationToken.checkPermission(asJob(), asJob().getAuthToken(), req, rsp);
 
         ParametersDefinitionProperty pp = asJob().getProperty(ParametersDefinitionProperty.class);
+        if (!asJob().isBuildable()) {
+            throw HttpResponses.error(SC_INTERNAL_SERVER_ERROR, new IOException(asJob().getFullName() + " is not buildable!"));
```

Comment:
Your point is correct, 409 response seems correct. I'll change it to that. 

Maybe the same should be also applied on doBuild() method?